### PR TITLE
ref: use getMetdata in renewal and register pages

### LINKED ISSUE #536: Don't ask for metadata again
## Description
Don't ask the email and tax state again to someone who already has metadata. 

## Proposed ToDo

- [ ] Use `getMetadata` on the `renewal.tsx` page
- [ ] Use `getMetadata` on the `autorenewal.tsx` page
- [ ] Use `getMetadata` on the `registerV2.tsx` page

### PR DESCRIPTION
**Pull request by irisdv**
Close #536

### CODE REVIEW

File: components/domains/renewal.tsx [hunk 1]
```diff
@@ -61,8 +58,9 @@ const Renewal: FunctionComponent<RenewalProps> = ({ groups }) => {
   const [termsBox, setTermsBox] = useState<boolean>(true);
   const [renewalBox, setRenewalBox] = useState<boolean>(true);
   const [walletModalOpen, setWalletModalOpen] = useState<boolean>(false);
-  const [salts, setSalts] = useState<string[] | undefined>();
-  const [metadataHashes, setMetadataHashes] = useState<string[] | undefined>();
```
Comment (fricoben) [inline components/domains/renewal.tsx:65]: Why we wouldn't need several metdatahashes anymore ? Is there a particular reason ? @Th0rgal might need you opinion here too

**Review by fricoben — CHANGES_REQUESTED** [review]
I have a question on the fact that you change the number of metadata we send for `renewal`

It seems strange cause I think we need several metadata, one for each renewal of the multicall. 

@Th0rgal Would love to have your opinion cause I'm not sure to see everything here

File: components/domains/renewal.tsx (hunk 1 above)
Comment (irisdv) [reply components/domains/renewal.tsx:65]: I asked @Th0rgal who told me one metahash was enough, if we want to use different metahash then I'd have to update the endpoint in the server to return an array with a metahash for each domain owned.

Comment (Th0rgal) [reply components/domains/renewal.tsx:65]: Yes one metadata should not be an issue, the endpoint must just return the last one

**Review by Th0rgal — APPROVED** [review]
lgtm

**Review by fricoben — APPROVED** [review]
lgtm

**Pull request merged**